Doc/library/asyncio-task.rst

```diff
@@ -1174,8 +1176,17 @@ Task Object
 
       Return the coroutine object wrapped by the :class:`Task`.
 
+      .. note::
+
+         This will return ``None`` for Tasks which have already
+         completed eagerly. See the :ref:`Eager Task Factory <eager-task-factory>`.
+
       .. versionadded:: 3.8
 
+      .. versionchanged:: 3.12
+
+         Added note on specific eager task execution behavior.
```

Comment:
I don't know if it makes sense calling out a documentation note as "added in 3.12"

Reply:
```suggestion
      .. versionchanged:: 3.12

        Newly added eager task execution means result may be ``None``.
```

Reply:
Okay, how about my proposed tweak.